source/adapters/cuda/adapter.cpp

```diff
@@ -57,7 +57,7 @@ UR_APIEXPORT ur_result_t UR_APICALL urAdapterRelease(ur_adapter_handle_t) {
 
 UR_APIEXPORT ur_result_t UR_APICALL urAdapterGetLastError(
     ur_adapter_handle_t, const char **ppMessage, int32_t *pError) {
-  std::ignore = pError;
+  *pError = ErrorMessageCode;
```

Comment:
`pError` is supposed to contain the adapter specific error message returned from CUDA/HIP/L0/OpenCL/etc. `ErrorMessageCode` is a `ur_result_t` so it shouldn't be getting used to set `pError`.